server-spi-private/src/main/java/org/keycloak/events/Details.java

```diff
@@ -100,4 +100,6 @@ public interface Details {
 
     String LOGOUT_TRIGGERED_BY_ACTION_TOKEN = "logout_triggered_by_action_token";
     String LOGOUT_TRIGGERED_BY_REQUIRED_ACTION = "logout_triggered_by_required_action";
+    String ACCESS_TOKEN_EXPIRATION_TIME = "access_token_expiration_time";
+    String AGE_OF_REFRESH_TOKEN = "age_of_refresh_token";
```

Comment:
Why we named this `AGE_OF_REFRESH_TOKEN` and not `AGE_OF_ACCESS_TOKEN`. Again, If I am not mistaken, they should be the same but I think we are more interested in the age access token and not refresh token.

Reply:
This event will be written when a token is refreshed. For this action, the input is the refresh token, and the output is the new access token. 

We are interested how many seconds ago this refresh token has been issues, therefore it is the "age" of the refresh token. And we want to know how long the new access token will be valid. 

Good case 1 - refreshing an access token shortly before it has expired: 
Refresh token age is 25 seconds, newly created access token will be valid for 30 seconds. 

Good case 2 - refreshing an access after the user came back from a coffee break: 
Refresh token age is 300 seconds, newly created access token will be valid for 30 seconds. 

Bad case - refreshing an access too early:
Refresh token age is 2 seconds, newly created access token will be valid for 30 seconds. 

Does that make sense?

Reply:
Yes, this makes sense. Still, people are/should be refreshing tokens because of the age of the access token, not the age of the refresh token which is irrelevant in this case.

To distinguish between good and bad cases we need to combine the two values and given the names `access_token_expiration_time` and `age_of_refresh_token`, it may not be obvious that these values "belong" together at first sight. 

What I meant is that if it was `access_token_age[_at_the_time_of_refresh]` or something similar, it may be more natural to understand why these two values are there.

Anyway, I think this is a great improvement already and I don't want to be nitpicking so I will approve this and I am ok to merge as is.

Reply:
ok, then let's start as is and see if we need to refine.